### PR TITLE
1233 port dashboard switch pattern

### LINKED ISSUE #1233: Combine dashboard theme switch pattern

### PR DESCRIPTION
**Pull request by richmccartney**
## Done

- Porting over the switch pattern from the dashboard theme
- Creates HTML and documentation for the new pattern

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- View the new pattern at http://0.0.0.0:8101/vanilla-framework/examples/patterns/switch/

## Details

Fixes #1233

### CODE REVIEW

File: scss/_patterns_switch.scss [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      color: $color-x-light;
+    }
+
+    :last-child {
+      background-color: $color-mid-light;
+      border-radius: 0 2px 2px 0;
+    }
+
+    &::before {
+      @extend .p-card--highlighted;
```
Comment (barrymcgee) [inline scss/_patterns_switch.scss]: Linting error: `@extend must be used with a %placeholder  placeholder-in-extend`

File: scss/_patterns_switch.scss [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      background-color: $color-x-light;
+      content: '';
+      display: block;
+      height: 100%;
+      left: 0;
+      max-height: 2rem;
```
Comment (barrymcgee) [inline scss/_patterns_switch.scss]: Please use [spacing variables](https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/_settings_spacing.scss)

File: scss/_patterns_switch.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    padding: initial;
+    position: relative;
+    width: 4rem;
+
+    &:focus {
+      outline: 1px solid $color-link;
+      outline-offset: 3px;
+    }
+
+    :first-child,
```
Comment (barrymcgee) [inline scss/_patterns_switch.scss]: Is this valid Sass without a parent (`&`) selector? Checking with a [quick codepen](https://codepen.io/barrymcgee/pen/rzraPo?editors=1100), I don't think it is?

File: scss/_patterns_switch.scss [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      background-color: $color-x-light;
+      content: '';
+      display: block;
+      height: 100%;
+      left: 0;
+      max-height: 2rem;
+      padding: 0;
+      position: absolute;
+      top: 0;
+      transition: .5s;
```
Comment (barrymcgee) [inline scss/_patterns_switch.scss]: Please use global [animation settings](https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/_settings_animations.scss).

File: examples/patterns/switch.html [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+     // Set click event to each switch control element
+     /**
+      * @param {String} switchBtn
+      */
+     switches.forEach(function (switchBtn) {
+       switchBtn.addEventListener('click', function(event) {
+
+         // If aria-checked is already true, set to false
+         if (this.getAttributeNode("aria-checked").value == 'false') {
```
Comment (barrymcgee) [inline examples/patterns/switch.html]: Can this be simplified by using a ternary operator? 

`el.setAttribute('aria-checked', el.getAttribute('aria-checked') === 'true' ? 'false' : 'true');`

File: examples/patterns/switch.html [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  <span>Off</span>
+</button>
+
+<h2>Example JavaScript</h2>
+<pre><code>
+(function() {
+  /**
+   * Control switch elements via toggling aria-checked
+   *
+   * @param {String} switchBtn
```
Comment (barrymcgee) [inline examples/patterns/switch.html]: We don't currently do anything with JSdoc comments and including it furthers adds to the volume and complexity of example code, can you remove please?

File: scss/_patterns_switch.scss (hunk 3 above)
Comment (richmccartney) [inline scss/_patterns_switch.scss]: Yeah this is valid you need to quickly look at what is written again. I've changed your example to make it work https://codepen.io/richmccartney/pen/BdPQGP?editors=1100

File: scss/_patterns_switch.scss (hunk 2 above)
Comment (richmccartney) [inline scss/_patterns_switch.scss]: Ah good spot ta

File: examples/patterns/switch.html (hunk 6 above)
Comment (richmccartney) [inline examples/patterns/switch.html]: I think form thje last meeting we are just leaving the JSDocs parts now.

**Pull request merged**